plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -200,18 +214,66 @@ def sweep(self, arguments: List[str]) -> None:
             returns = self.launcher.launch(overrides, initial_job_idx=self.job_idx)
             self.job_idx += len(returns)
             for trial, ret in zip(trials, returns):
-                study._tell(trial, optuna.trial.TrialState.COMPLETE, [ret.return_value])
+                values: Optional[List[float]] = None
+                state: optuna.trial.TrialState = optuna.trial.TrialState.COMPLETE
+                try:
+                    if len(directions) == 1:
+                        try:
+                            values = [float(ret.return_value)]
+                        except (ValueError, TypeError):
+                            raise ValueError(
+                                f"Return value must be float-castable. Got '{ret.return_value}'."
+                            )
+                    else:
+                        if not isinstance(ret.return_value, (list, tuple)):
+                            raise ValueError(
+                                "Return value must be a list or tuple of float-castable values."
+                                f" Got '{ret.return_value}'."
+                            )
+                        try:
+                            values = [float(v) for v in ret.return_value]
+                        except (ValueError, TypeError):
+                            raise ValueError(
+                                "Return value must be a list or tuple of float-castable values."
+                                f" Got '{ret.return_value}'."
```

Comment:
Thank you for pointing it out.
I think `if not isinstance(ret.return_value, (list, tuple))` is not required because the try-except from L233-L239 raises the same error without it. I removed it in the commit https://github.com/facebookresearch/hydra/pull/1294/commits/7d1b40fc0e11631bad01eb144e1e8f97f39b58cd.